Model/User.hs

```diff
@@ -31,13 +36,16 @@ findUsers userIds = selectList [UserId <-. userIds] []
 findUsers' :: [UserId] -> DB [User]
 findUsers' = fmap (map entityVal) . findUsers
 
-authenticateUser :: Creds m -> DB (Maybe UserId)
+authenticateUser :: AuthId m ~ UserId => Creds m -> DB (AuthenticationResult m)
```

Comment:
It has to do with type families, see this comment: https://github.com/thoughtbot/carnival/pull/205#discussion_r27682987